### PR TITLE
[stdlib] Remove FileCheck for `test_simd.mojo`

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Related to https://github.com/modularml/mojo/issues/2024

Note that https://github.com/modularml/mojo/pull/2279 should be merged first, then we can rebase this PR, it's going to be cleaner.

### CODE REVIEW

File: stdlib/test/builtin/test_simd.mojo
Comment (JoeLoser) [inline stdlib/test/builtin/test_simd.mojo:43]: **Suggestion** How about we actually use `SIMD.eq` rather than `str()` on both sides essentially?  `str` was only there for `FileCheck` since it's just a fancy regex matcher in short.  We can actually call real methods and not just do string matching shenanigans now that we're using the `testing` module.

**Review by JoeLoser — APPROVED** [review]
Looks good, thank you! Just one minor comment in a test.  Do you want to rebase now that https://github.com/modularml/mojo/commit/ee5dfe791effa9def469001e9cd5933996b96c6f landed, or want me to do "rebase and merge" after the test fix?

File: stdlib/test/builtin/test_simd.mojo
Comment (gabrieldemarmiesse) [reply stdlib/test/builtin/test_simd.mojo:43]: For this case in particular, I don't think switching would make sense. We would end up with 
```
assert_equal(SIMD[DType.index, 4](52, 12, 43, 5), SIMD[DType.index, 4](52, 12, 43, 5))
```
which seems kind of pointless as a test. To verify that the numbers are correctly stored in the SIMD, I think converting to string is a valid thing to do

**Comment by gabrieldemarmiesse** [comment]
I merged nightly into this branch. I believe you can "squash and merge" @JoeLoser

File: stdlib/test/builtin/test_simd.mojo
Comment (JoeLoser) [reply stdlib/test/builtin/test_simd.mojo:43]: Oh, I see now - thanks for pointing that out.  The whole test seems pretty low value, honestly, but we can keep it for now; not a biggie.  We're in the process of syncing repos right now to push out a new `nightly/mojo`, so I'm holding off from merging this, but I'll merge it later today.

Comment (gabrieldemarmiesse) [reply stdlib/test/builtin/test_simd.mojo:43]: Sure thing!

**Comment by JoeLoser** [comment]
@gabrieldemarmiesse I think you'll need to rebase again from more conflicts when we synced the repos/made a new `nightly/mojo` just a bit ago.  Once that's done, I'll merge this PR.  Thanks!

**Comment by gabrieldemarmiesse** [comment]
Haha, we'll merge this PR eventually XD

**Comment by JoeLoser** [comment]
> Haha, we'll merge this PR eventually XD

Merged!

**Pull request merged**
